memote/support/basic.py

```diff
@@ -291,15 +291,14 @@ def find_duplicate_reactions(model):
     duplicates = []
     rxn_db_identifiers = ["metanetx.reaction", "kegg.reaction", "brenda",
                           "rhea", "biocyc", "bigg.reaction"]
-    ann_rxns = [(rxn, rxn.annotation) for rxn in model.reactions
-                for db in rxn_db_identifiers if db in rxn.annotation]
-    for a, b in combinations(ann_rxns, 2):
-        for db in rxn_db_identifiers:
-            try:
-                if a[1][db] == b[1][db]:
-                    duplicates.append((a[0], b[0]))
-            except KeyError:
-                continue
+    ann_rxns = [(rxn, frozenset((key, rxn.annotation[key])
```

Comment:
Good thinking adding the condition :smiley: The only thing missing now is how to handle the case when there are multiple identifiers for one database. I'm not 100% sure now, that definitely happens for metabolites but maybe not for reactions? If it does happen, you'd have to add another small loop.